robot-command/src/main/java/org/obolibrary/robot/ExplainCommand.java

```diff
@@ -0,0 +1,146 @@
+package org.obolibrary.robot;
+
+import java.io.File;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Options;
+import org.semanticweb.owl.explanation.api.Explanation;
+import org.semanticweb.owlapi.apibinding.OWLManager;
+import org.semanticweb.owlapi.manchestersyntax.parser.ManchesterOWLSyntaxInlineAxiomParser;
+import org.semanticweb.owlapi.model.IRI;
+import org.semanticweb.owlapi.model.OWLAnnotationAssertionAxiom;
+import org.semanticweb.owlapi.model.OWLAxiom;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
+import org.semanticweb.owlapi.util.SimpleShortFormProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles inputs and outputs for the {@link ExplainOperation}.
+ *
+ * @author <a href="mailto:balhoff@renci.org">Jim Balhoff</a>
+ */
+public class ExplainCommand implements Command {
+  /** Logger. */
+  private static final Logger logger = LoggerFactory.getLogger(ExplainCommand.class);
+
+  /** Store the command-line options for the command. */
+  private Options options;
+
+  public ExplainCommand() {
+    Options o = CommandLineHelper.getCommonOptions();
+    o.addOption("i", "input", true, "load ontology from a file");
+    o.addOption("I", "input-iri", true, "load ontology from an IRI");
+    o.addOption("r", "reasoner", true, "reasoner to use: ELK, HermiT, JFact");
+    o.addOption("a", "axiom", true, "the axiom to explain");
+    o.addOption("m", "max", true, "the maximum number of explanations to retrieve");
+    o.addOption("f", "format", true, "the explanation result format: MD, OFN, etc.");
+    o.addOption("o", "output", true, "save explanation to a file");
```

Comment:
Just a thought...

Instead of having to chain to `convert` after using `explain`, the `--output` option could output the ontology just containing explanation options in any valid ontology format. And you could have an `--explanation` option for the markdown or text file, e.g.:
```
o.addOption("o", "output", true, "save explanation axioms to an ontology");
o.addOption("e", "explanation", true, "save explanation to a file");
```
For the other ROBOT commands, the `--output` is an ontology. For example, `query` does not use the `--output` flag for the generated results.

Reply:
Agree.